### DISCUSSION TITLE
Firmware versions of the devices

### DISCUSSION THREAD

**Discussion by Flo-Schilli** [Flo-Schilli/homeassistant-grohe_sense#27] [General]
Hi all,

it would be really interessting for me what different Firmware Versions are there out in the field of the different devices. As with the different versions the devices do - or don't - have some specific features. 

Here are mine:

Grohe Sense Guard: `03.06`
Grohe Sense: `01.20`
Grohe Blue Home: `N/A`
Grohe Blue Professional: `N/A`

**Comment by daxyorg** [comment]
Grohe Sense Guard `22502LN1`: `02.05` [_OTA update after inquiry to Grohe Support_]
Grohe Sense: `01.20`
Grohe Blue Home: `01.04` `02.10.Z10.0300.0104` (Aug.2025)
Grohe Blue Professional: `N/A`

**Comment by vincedarley** [comment]
Grohe Sense Guard 02.05 (updated to this automatically when I installed the device a month ago).
No other Grohe devices.

**Comment by rama1981** [comment]
Grohe Sense Guard: N/A
Grohe Sense: N/A
Grohe Blue Home: N/A
Grohe Blue Professional: 1.04.Z10.0300.0104

**Reply by daxyorg** [reply]
Thank you! It was useful for me "to set my watch".

**Comment by Flo-Schilli** [comment]
I'm really curious why I do have `03.06` for the Sense Guard.
@daxyorg would you like to test, if the stuff with the pressure measurement is working in 02.06? If yes, I would create a separate branch for that in a couple of days.

**Reply by daxyorg** [reply]
Yes, of course, with pleasure I'll be test it, but I'll be at home after next Wednesday (Nov, 23th).

**Reply by szappanyos** [reply]
I could try with 02.05

**Reply by Flo-Schilli** [reply]
I just created a new branch called testing/pressure-measurement where the minimum version for the Grohe Sense must be 02.05 so you both should be able to test. 
If the new sensors stay on 'Unknown' even after triggering the pressure measurement button and waiting for max 5min, then it is not working with that version.
So any feedback is welcome :-)

**Reply by daxyorg** [reply]
Hi, after some delay, I opened this ([https://github.com/Flo-Schilli/homeassistant-grohe_sense/issues/33])

**Comment by szappanyos** [comment]
Grohe Sense Guard: 02.05
Grohe Sense: 01.36 and 1.40
Grohe Blue Home: N/A
Grohe Blue Professional: N/A

**Comment by yoali11** [comment]
Grohe Sense Guard: 02.05
Grohe Sense: 01.40
Grohe Blue Home: N/A
Grohe Blue Professional: N/A

**Comment by patricknitsch** [comment]
Grohe Sense 1.40
Grohe Sense Guard 2.05

**Comment by skorpi0n** [comment]
Grohe Sense Guard: 01.44
Grohe Sense: 01.06

I cant see anywhere in the app (v2.2.1) where I can update the firmware on the Sense Guard. I have auto-update enabled. Anyone know how to do it?

**Reply by skorpi0n** [reply]
I just modified the version check in sensor.py and button.py and restarted HA (should that be enough?) and all new entities remained in "unknown" state even after waiting over 5 minutes after triggering the pressure measurement button.
So version 1.44 definitely doesn't work

**Reply by Flo-Schilli** [reply]
Hi @skorpi0n 👋 

I think if you're not seeing the updates you have to contact the support from Grohe. I think that's how others got an update available. I don't know why Grohe is not pushing it more to the devices tbh.
Without looking into the source I think you modified the two correct files. Also there is a separate branch for testing with 02.XX - Version of the device. Also thanks for your feedback that it's definitely not working with your version. I think I have to note it down somewhere 😊

**Reply by daxyorg** [reply]
Yes, @skorpi0n , open a ticket https://www.sensehelp.com/ and from Grohe will push you version `02.05` with instructions.

**Reply by skorpi0n** [reply]
So, I contacted Grohe for a firmware update request but the update failed, which left me with a device that I could connect to my wifi and ping, but the device didn't connect to Grohe.
Grohe kindly sent me a replacement device with updated firmware (02.05)
To complete the warranty process I had to send back the old device to Grohe

As the firmware update from 01.44 to 02.05 didn't give the new features with pressure measurement, I would advice people from updating to a newer firmware if they don't absolute need it. But maybe it was just for me the update failed.

**Comment by daxyorg** [comment]
Grohe sense guard has 2 different versions with different EAN numbers.
This may be is the reason why there are 2 updates (v 02.05 and 03.06). I don't know how to check but here it is the versions:

22502LN0 with EAN 4005176412066

22502LN1 with EAN 4005176477256

**_This is just my guess._**


https://www.grohe.de/de_de/grohe-sense-set-1x-grohe-sense-guard-3x-grohe-sense-22502LN0.html


https://www.grohe.de/de_de/grohe-sense-set-1x-grohe-sense-guard-3x-grohe-sense-22502LN1.html
